grep-0018-modernize-c++.md

```diff
@@ -0,0 +1,417 @@
+# GREP 0018 -- Modernize C++
+
+- Original Author: Thomas Habets <thomas@habets.se>
+- Champion: Thomas Habets <thomas@habets.se>
+- Status: Draft
+
+History:
+- 20-Jan-2020: Initial Draft
+
+## Abstract
+
+As of early 2020 there is lots of pre-C++11 code in GNU Radio, and
+some other pattern of non-idiomatic C++.
+
+This GREP lays out how to get from here to there with minimal
+breaking.
+
+## Copyright / License
+
+This GREP is licensed as CC-BY-ND. Copyright 2020 Google Inc.
+
+## Motivation
+
+Cleaning up whole patterns of the code 
+
+The legacy patterns this GREP lays out a plan for making the code base:
+
+* more readable: standard components, less code, and explicit const
+* safer: less manual memory management, prevent inherently broken copy
+* faster: fewer pointer indirections, more information to the compiler
+
+## Description
+
+### Open questions
+
+* What level of ABI guarantee does GNU Radio make between and within
```

Comment:
This is an answered question, it's why we have semver. We check for ABI breakages on stable releases, and our version numbers state tell us if ABI was changed or not.

Reply:
Yeah, added this before asking on the list, removing.